src/Ormolu/Utils/IO.hs

```diff
@@ -0,0 +1,19 @@
+-- | Write 'Text' to files using UTF8 and ignoring native
+-- line ending conventions.
+module Ormolu.Utils.IO
+  ( writeFileUtf8,
+    readFileUtf8,
+  )
+where
+
+import Control.Exception (throwIO)
+import Control.Monad.IO.Class
+import qualified Data.ByteString as B
+import Data.Text (Text)
+import qualified Data.Text.Encoding as TE
+
+writeFileUtf8 :: MonadIO m => FilePath -> Text -> m ()
+writeFileUtf8 p = liftIO . B.writeFile p . TE.encodeUtf8
+
+readFileUtf8 :: MonadIO m => FilePath -> m Text
+readFileUtf8 p = liftIO $ either throwIO pure . TE.decodeUtf8' =<< B.readFile p
```

Comment:
Since you're throwing anyway, why not use `TE.decodeUtf8`?

Reply:
I did this out of habit to avoid "hidden" exceptions in lazy values (i.e. with `TE.decodeUtf8`, `void $ readFileUtf8 /path/to/existing/nonutf8file` wil succeed), but I can also change it if this is not considered to be important enough.